File: dev/20190701/k2works/api/rental/src/main/java/rental/Customer.java
package rental;

import java.util.ArrayList;
import java.util.List;

class Customer {
  String name;
  List<Rental> rentals = new ArrayList<>();

  Customer(String name) {
    this.name = name;
  }

  void addRental(Rental rental) {
    rentals.add(rental);
  }

  String name() {
    return name;
  }

  String statement() {
    int totalAmount = 0;
    int points = 0;

    String result = name() + "様のレンタル明細\n";

    for (Rental each : rentals) {
      int thisAmount = 0;
      switch (each.movie().priceCode()) {
        case Movie.REGULAR:
          thisAmount += 200;
          if (each.daysRented() > 2) thisAmount +=
            (each.daysRented() - 2) * 150;
          break;
        case Movie.NEW_RELEASE:
          thisAmount += each.daysRented() * 300;
          break;
        case Movie.CHILDREN:
          thisAmount += 150;
          if (each.daysRented() > 3) thisAmount +=
            (each.daysRented() - 3) * 150;
      }

      points++;
      if (
        each.movie().priceCode() == Movie.NEW_RELEASE && each.daysRented() > 1
      ) points++;

      result += "\t" + each.movie().title() + "\t" + thisAmount + "円\n";

      totalAmount += thisAmount;
    }

    result += "合計金額 " + totalAmount + "円\n";
    result += "獲得ポイント " + points + "p";
    return result;
  }
}

File: dev/20190710/k2works/api/rental/customer.py
from .movie import Movie


class Customer:
    def __init__(self, name):
        self.__name = name

        self.__rentals = []

    @property
    def name(self):
        return self.__name

    def add_rental(self, arg):
        self.__rentals.append(arg)

    def statement(self):
        total_amount, frequent_renter_points = 0, 0

        result = f"Rental Record for {self.name}\n"

        for element in self.__rentals:
            this_amount = 0

            # 各行の金額を計算
            if element.movie.price_code == Movie.REGULAR:
                this_amount += 2
                if element.days_rented > 2:
                    this_amount += (element.days_rented - 2) * 1.5
            elif element.movie.price_code == Movie.NEW_RELEASE:
                this_amount += element.days_rented * 3
            elif element.movie.price_code == Movie.CHILDRENS:
                this_amount += 1.5
                if element.days_rented > 3:
                    this_amount += (element.days_rented - 3) * 1.5

            # レンタルポイントを加算
            frequent_renter_points += 1
            # 新作2日間レンタルでボーナス点を加算
            if element.movie.price_code == Movie.NEW_RELEASE and element.days_rented > 1:
                frequent_renter_points += 1
            # このレンタルの料金を表示
            result += f"\t{element.movie.title}\t{str(this_amount)}\n"
            total_amount += this_amount
        # フッター行を追加
        result += f"Amount owed is {total_amount}\n"
        result += f"You earned {frequent_renter_points} frequent renter points"
        return result
